src/NServiceBus.Core/Transports/Learning/DirectoryBasedTransaction.cs

```diff
@@ -104,10 +104,25 @@ void RecoverPending()
             //only need to move the incoming file
             foreach (var file in pendingDir.EnumerateFiles(TxtFileExtension))
             {
-                File.Move(file.FullName, Path.Combine(basePath, file.Name));
+                var destFileName = Path.Combine(basePath, file.Name);
+                try
+                {
+                    File.Move(file.FullName, destFileName);
+                }
+                catch (IOException e)
+                {
+                    log.Debug($"Unable to move pending transaction from '{file.FullName}' to '{destFileName}'. Pending transaction is assumed to be recovered by a competing consumer.", e);
+                }
             }
 
-            pendingDir.Delete(true);
+            try
+            {
+                pendingDir.Delete(true);
+            }
+            catch (IOException e)
+            {
+                log.Debug($"Unable to delete pending transaction directory '{pendingDir.FullName}'.", e);
```

Comment:
Should we add retries similar to https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Transports/Learning/AsyncDirectory.cs ?